### PR TITLE
python3Packages.jinjax: init at 0.48

### LINKED ISSUE #379399: Package request: python313Packages.jinjax
### Nixpkgs version

- Unstable (25.05)

### Describe the package

JinjaX is a Python library for creating reusable "components": encapsulated template snippets that can take arguments and render to HTML. They are similar to React or Vue components, but they render on the server side, not in the browser.

### Upstream homepage

https://jinjax.scaletti.dev/

### Source URL

https://github.com/jpsca/jinjax

### License

MIT

### Platforms

x86_64-linux

### Additional context

_No response_

### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this package does not yet exist in an [open pull request](https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+is%3Apr+label%3A%228.has%3A+package+%28new%29%22) or in [Nixpkgs Unstable](https://search.nixos.org/packages?channel=unstable).
- [x] I assert that this is not a [duplicate of any known issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+label%3A%220.kind%3A+packaging+request%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### For this package's maintainers:

Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by ethancedwards8**
## Things done

Closes: https://github.com/NixOS/nixpkgs/issues/379399

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ethancedwards8** [comment]
```
> HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
       > _____________________ ERROR collecting src/jinjax/utils.py _____________________
       > import file mismatch:
       > imported module 'jinjax.utils' has this __file__ attribute:
       >   /nix/store/m54jvscasryawqnr9xc5v4r2whhwzyk5-python3.12-jinjax-0.48/lib/python3.12/site-packages/jinjax/utils.py
       > which is not the same as the test file we want to collect:
       >   /private/tmp/nix-build-python3.12-jinjax-0.48.drv-0/source/src/jinjax/utils.py
       > HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
       > =========================== short test summary info ============================
       > ERROR benchmark/benchmark.py
       > ERROR benchmark/profile.py
       > ERROR docs/docs.py
       > ERROR src/jinjax/catalog.py
       > ERROR src/jinjax/component.py
       > ERROR src/jinjax/exceptions.py
       > ERROR src/jinjax/html_attrs.py
       > ERROR src/jinjax/jinjax.py
       > ERROR src/jinjax/middleware.py
       > ERROR src/jinjax/utils.py
```
I already know I'll get a question about pytest hook. Anyone have any ideas?

File: pkgs/development/python-modules/jinjax/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  build-system = [ poetry-core ];
+
+  dependencies = [
+    jinja2
+    markupsafe
+  ];
+
+  pythonImportsCheck = [ "jinjax" ];
+
```
[the remark this answers is not in the document]
Comment (ethancedwards8) [reply pkgs/development/python-modules/jinjax/default.nix:30]:
```
> checking for references to /private/tmp/nix-build-python3.12-jinjax-0.48.drv-0/ in /nix/store/kg71n0zxai7pyna5vzs8w97piq798yfs-python3.12-jinjax-0.48-dist...
       > patching script interpreter paths in /nix/store/kg71n0zxai7pyna5vzs8w97piq798yfs-python3.12-jinjax-0.48-dist
       > Executing pythonRemoveTestsDir
       > Finished executing pythonRemoveTestsDir
       > Running phase: installCheckPhase
       > Executing versionCheckPhase
       > versionCheckHook: /nix/store/baif0bgjnj3m1k4jnn2yfgggdwm9b6zm-python3.12-jinjax-0.48/bin/jinjax was not found, or is not an executable
       For full logs, run 'nix log /nix/store/fs0n53z3jgyd30rjs0nz9wsfjy0ap6mv-python3.12-jinjax-0.48.drv'.
```
I don't think `versionCheckHOok` applies for a python library, does it?

[the remark this answers is not in the document]
Comment (ethancedwards8) [reply pkgs/development/python-modules/jinjax/default.nix:30]: See the above comment (outside of review thread) please. I had issues with getting it to work

**Comment by ethancedwards8** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 384432`

---
### `x86_64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.jinjax</li>
    <li>python312Packages.jinjax.dist</li>
    <li>python313Packages.jinjax</li>
    <li>python313Packages.jinjax.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/jinjax/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (ethancedwards8) [reply pkgs/development/python-modules/jinjax/default.nix:31]: There aren't any in the repo. Not sure what to do.

**Comment by ethancedwards8** [comment]
Gonna just close this. I don't need it, was just trying to be helpful. If someone needs it, feel free to use this as a base.
